context/ifc-context.tsx

```
@@ -457,42 +457,64 @@ export function IFCContextProvider({ children }: { children: ReactNode }) {
-            } else if (itemProps) {
-              for (const key in itemProps) {
-                if (
-                  Object.prototype.hasOwnProperty.call(itemProps, key) &&
-                  typeof itemProps[key] === "object" &&
-                  itemProps[key] !== null &&
-                  itemProps[key].Name?.value === psetName &&
-                  itemProps[key].HasProperties // Check if it looks like a PSet
-                ) {
-                  psetObject = itemProps[key];
+            // Try direct access via itemProps.propertySets first
+            if (itemProps && itemProps.propertySets) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)